lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -124,6 +122,38 @@ SdfLayerHandle defPrimSpecLayer(const UsdPrim& prim)
 	return defLayer;
 }
 
+bool isTargetLayerHaveOpinion(const UsdPrim& prim)
+{
+    auto editTarget = prim.GetStage()->GetEditTarget();
+    auto layer = editTarget.GetLayer();
+    auto primSpec = layer->GetPrimAtPath(prim.GetPath());
+
+    // to know whether the target layer contains any opinions that 
+    // affect a particular prim, there must be a primSpec for that prim
+    if (!primSpec) {
+        return false;
+    }
+
+    return true;
+}
+
+SdfLayerHandle targetLayerWithOpion(const UsdPrim& prim)
```

Comment:
Typo: "Opion" --> "Opinion".
Also, isn't it the case that this will return the highest-priority layer that has an opinion, either def or over?  It would be good if the function would say that, e.g.
SdfLayerHandle strongestLayerWithOpinion(const UsdPrim& prim)
or
SdfLayerHandle strongestLayerWithPrimSpec(const UsdPrim& prim)


A minor point is that the existing function in the Utils file is called defPrimSpecLayer, i.e. uses "prim spec" rather than "opinion".  Perhaps we should standardize on one terminology or the other.

Reply:
Please see 54166d2b6

I agree with standardization on one terminology, I picked "prim spec" instead of "opinion" for now.
Regarding the function names, I made following changes and dropped the target. Let me know if this reads better:

**Before:**
```
isTargetLayerHaveOpinion 
targetLayerWithOpion      
```
**After:**
```
doesLayerHavePrimSpec  
strongestLayerWithPrimSpec
```